alicloud/resource_alicloud_cen_route_entry_publishment_test.go

```diff
@@ -0,0 +1,180 @@
+package alicloud
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/cbn"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAlicloudCenRouteEntryPublishment_basic(t *testing.T) {
+	var routeEntry cbn.PublishedRouteEntry
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+		},
+
+		// module name
+		IDRefreshName: "alicloud_cen_route_entry_publishment.foo",
+		Providers:     testAccProviders,
+		CheckDestroy:  testAccCheckCenRouteEntryPublishmentDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccCenRouteEntryPublishmentConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCenRouteEntryPublishmentExists("alicloud_cen_route_entry_publishment.foo", &routeEntry),
+					resource.TestCheckResourceAttr("alicloud_cen_route_entry_publishment.foo", "cidr_block", "11.0.0.0/16"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckCenRouteEntryPublishmentExists(n string, routeEntry *cbn.PublishedRouteEntry) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No Cen Route Entry Publishment ID is set")
+		}
+
+		client := testAccProvider.Meta().(*AliyunClient)
+
+		var para CenRouteEntryPublishmentParas
+		para.cenId = rs.Primary.Attributes["instance_id"]
+		para.childInstanceId = rs.Primary.Attributes["child_instance_id"]
+		para.cidr = rs.Primary.Attributes["cidr_block"]
+		para.regionId = rs.Primary.Attributes["region_id"]
+		para.vtbId = rs.Primary.Attributes["route_table_id"]
+
+		routeEntryItem, err := client.DescribePublishedRouteEntriesById(para)
+		if err != nil {
+			return err
+		}
+
+		if routeEntryItem.PublishStatus != string(PUBLISHED) {
+			return fmt.Errorf("CEN id %s child instance id %s vtb id %s cidr %s: route entry publishment status error", para.cenId, para.childInstanceId, para.vtbId, para.cidr)
+		}
+
+		*routeEntry = routeEntryItem
+		return nil
+	}
+}
+
+func testAccCheckCenRouteEntryPublishmentDestroy(s *terraform.State) error {
+	client := testAccProvider.Meta().(*AliyunClient)
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "alicloud_cen_route_entry_publishment" {
+			continue
+		}
+
+		var para CenRouteEntryPublishmentParas
+		para.cenId = rs.Primary.Attributes["instance_id"]
+		para.childInstanceId = rs.Primary.Attributes["child_instance_id"]
+		para.cidr = rs.Primary.Attributes["cidr_block"]
+		para.regionId = rs.Primary.Attributes["region_id"]
+		para.vtbId = rs.Primary.Attributes["route_table_id"]
+
+		routeEntryItem, err := client.DescribePublishedRouteEntriesById(para)
+		if err != nil {
+			if NotFoundError(err) || IsExceptedError(err, ParameterIllegalCenInstanceId) {
```

Comment:
I think `IsExceptedError(err, ParameterIllegalCenInstanceId)` is redundant and `NotFoundError(err)` contains it.